Anonymize paths in `fastlane env` output

0xced opened the pull request:
- [x] Run `bundle exec rspec` from the subdirectory of each tool you modified. Alternatively, run `rake test_all` from the root directory.
- [x] Run `bundle exec rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

---

This way, the username is not leaked. Also, it makes it easy to visually spot when a path is in the user’s home directory.

0xced committed:
Anonymize paths in `fastlane env` output

This way, the username is not leaked. Also, it makes it easy to visually spot when a path is in the user’s home directory.

hjanuschka commented:
i really like that. recently was thinking about doing a `UI.input`  before copy&paste - and asking the user for strings that they wan't to be removed? like (app identifier ...) - what do you think about it @0xced

0xced commented:
@hjanuschka This is a good idea, but it should be done in another pull request IMHO. I wonder how much data we could anonymize automatically, without needing `UI.input`.

milch commented on fastlane/lib/fastlane/environment_printer.rb:
```diff
@@ -270,6 +270,10 @@ def self.print_fastlane_files
     end
 
+    def self.anonymized_path(path)
+      return ENV['HOME'] ? path.gsub(/^#{ENV['HOME']}/, '~') : path
```
In the super edgy edge case that the user's home dir is the prefix of other folders this is gonna output garbage. Example: `$HOME=/work`, `anonymized_path("/workspace/whatever/fastlane")` => `~space/whatever/fastlane`. 

Might be more applicable to linux where the user's home dir is not generally inside the `/Users/` folder.

hjanuschka commented:
@milch @0xced 

how about this PR here -> https://github.com/fastlane/fastlane/pull/5982
we could refactor all options and set the flags on all critical parameters, and use this in some way to anonymize?

milch commented:
@hjanuschka Yeah, I've thought about something like that before. Sounds good to me, @0xced do you have any thoughts about that?

0xced commented:
I updated the code, and added a [spec](https://github.com/0xced/fastlane/blob/8a93790d8ecee25e9b6b8e83499592b6e7443c41/fastlane/spec/env_spec.rb#L39-L42).

With $HOME == /Users/john
`/Users/john/.fastlane/bin/bundle/bin/fastlane` → `~/.fastlane/bin/bundle/bin/fastlane`
`/Users/john` → `~`
`/Users/john/` → `~/`

With $HOME == /work
`/workspace/project/test` → `/workspace/project/test`

milch reviewed (approved):
Looks good now! Thanks for your contribution 🚀
